geowebcache/wmts/src/main/java/org/geowebcache/service/wmts/WMTSGetCapabilities.java

```diff
@@ -73,14 +74,25 @@ public class WMTSGetCapabilities {
 
     private final Collection<WMTSExtension> extensions;
 
+    private SecurityDispatcher securityDispatcher;
```

Comment:
What's a SecurityDispatcher? It's not included in the PR.

Thinking out loud, why don't you work on the TileLayerDispatcher instead, it's sort of a Catalog equivalent, its contents might be filtered on the fly based on a... maybe a pluggable TileLayerDispatcherFilter (similar to the CatalogFilter), that would avoid changing the WMTS code and would also make it support all tiled layer services, rather than having WMTS respond one way, and TMS, WMS-C, VE and GMAPS in another...

Reply:
I like that idea;

a) will create a TileLayerDispatcherFilter interface
b) provide implementation- as per the code in WMTSGetCapabilities
c) add filtering support in TileLayerDispatcher#getLayers()

Reply:
thanks!

Reply:
Just getting ready to commit this.  
I had to add a `getLayersFiltered()` method, since the API contract was to include all layer.  There was code (i.e. for attaching/removing listeners as well as other) that assumed it was getting all the layers.

Reply:
committed and added test cases

Reply:
Looks like it is a much better approach, marking conversation as resolved.